internal/http/client.go

```diff
@@ -0,0 +1,46 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras/internal/credential"
+	"oras.land/oras/internal/trace"
+	"oras.land/oras/internal/version"
+)
+
+// client option struct
+type ClientOptions struct {
+	Credential      auth.Credential
```

Comment:
Just out of curiosify, could you help me explain the difference between the Credential and CredentialStore.